src/idl_gen_text.cpp

```diff
@@ -144,6 +159,27 @@ bool Print<const void *>(const void *val, Type type, int indent,
         // clang-format on
       }
       break;
+    case BASE_TYPE_ARRAY: {
```

Comment:
Above, BASE_TYPE_VECTOR redefine `type` to `type = type.VectorType();` and use it as an element type.
Better rewrite BASE_TYPE_ARRAY branch in the uniform style.
Maybe introduce `elem_type` in both branches.